### DISCUSSION TITLE
Migration from Ipsec.conf to swanctl

### DISCUSSION THREAD

**Discussion by charmnQuark** [strongswan/strongswan#1911] [Q&A] [answered]
I am confused about the process of migration. ( for reference: I used this [guide](https://www.digitalocean.com/community/tutorials/how-to-set-up-an-ikev2-vpn-server-with-strongswan-on-ubuntu-20-04) to deploy) 

I have followed this [guide](https://wiki.strongswan.org/projects/strongswan/wiki/Fromipsecconf) . And ran the script to automate the process. I replace

**OS**: Ubuntu VERSION="20.04.3 LTS (Focal Fossa)"
**swanctl** : strongSwan swanctl 5.8.2
 **ipsec**: Linux strongSwan U5.8.2/K5.4.0-162-generic

However it does not say, if I should empty the ipsec.conf file, Or modify the service start etc.  May you please point me to the idiots guide.

Kind regards

**Comment by drcyberg** [comment]
Why don't you use the example configurations? [LINK](https://docs.strongswan.org/docs/5.9/config/IKEv2.html)

**Reply by charmnQuark** [reply]
Thank you for your reply,

That is a good idea, I was looking at the [console.log](https://www.strongswan.org/testing/testresults/ikev2/net2net-psk/) It uses, "strongswan" service.

`systemctl start strongswan`

however that service is not defined, based on the guide from Digitial ocean I had manually created the service.

```
root@vpn-server:~# service ipsec status
● strongswan-starter.service - strongSwan IPsec IKEv1/IKEv2 daemon using ipsec.conf
     Loaded: loaded (/lib/systemd/system/strongswan-starter.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2023-09-16 22:06:26 UTC; 1 day 10h ago
   Main PID: 680 (starter)
      Tasks: 18 (limit: 1116)
     Memory: 9.1M
     CGroup: /system.slice/strongswan-starter.service
             ├─680 /usr/lib/ipsec/starter --daemon charon --nofork
             └─735 /usr/lib/ipsec/charon

**commented out** 



root@vpn-server:~# cat /lib/systemd/system/strongswan-starter.service
[Unit]
Description=strongSwan IPsec IKEv1/IKEv2 daemon using ipsec.conf
After=network-online.target

[Service]
ExecStart=/usr/sbin/ipsec start --nofork
ExecReload=/usr/sbin/ipsec reload
StandardOutput=syslog
Restart=on-abnormal

[Install]
WantedBy=multi-user.target
```

**Reply by tobiasbrunner** [reply, marked as answer]
I'd actually recommend the [quick start configs](https://docs.strongswan.org/docs/5.9/config/quickstart.html), rather than the configs from the testing environment (see [here](https://docs.strongswan.org/docs/5.9/config/configExamplesNotes.html) why using those might not be such a good idea).

Also, the service you are using (starter/ipsec) is the old one that is configured via ipsec.conf (i.e. if you use that there is no need for any config migration). The `strongswan` service is the newer one (installed on Debian/Ubuntu via `charon-systemd` package). If you want to use the latter, make sure you disable or uninstall the other one (the package is `strongswan-starter`).

**Reply by charmnQuark** [reply]
I have starter/ipsec installed successfully, ( and it has been working for the last few years). I have a new use case whereby the ip address are identical on both sides of the VPN . I found this use case/ documentation (  https://www.strongswan.org/testing/testresults/ikev2/net2net-same-nets/ ) but since its using the newer service, I had thought of migrating.

I have set up a new test vm, and I will try your advise of installing the new package. I will provide feedback. Thanks again.

**Comment by charmnQuark** [comment]
> I'd actually recommend the [quick start configs](https://docs.strongswan.org/docs/5.9/config/quickstart.html), rather than the configs from the testing environment (see [here](https://docs.strongswan.org/docs/5.9/config/configExamplesNotes.html) why using those might not be such a good idea).
> 
> Also, the service you are using (starter/ipsec) is the old one that is configured via ipsec.conf (i.e. if you use that there is no need for any config migration). The `strongswan` service is the newer one (installed on Debian/Ubuntu via `charon-systemd` package). If you want to use the latter, make sure you disable or uninstall the other one (the package is `strongswan-starter`).


Thank you very much, this worked.
